pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5673,3 +5673,157 @@ func TestRhdmProdImmutableEnvironmentWithoutJbpmCluster(t *testing.T) {
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.True(t, len(cr.Spec.Objects.Servers[0].Kafka.Topics) == 2)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+
+	for _, env := range envs {
+
+		switch e := env.Name; e {
+		case "KIE_SERVER_KAFKA_EXT_ENABLED":
+			assert.True(t, env.Value == "true")
+
+		case "KIE_SERVER_KAFKA_EXT_GROUP_ID":
+			assert.True(t, env.Value == "my-kafka-group")
+
+		case "KIE_SERVER_KAFKA_EXT_ACKS":
+			assert.True(t, env.Value == "2")
+
+		case "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS":
+			assert.True(t, env.Value == "true")
+
+		case "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS":
+			assert.True(t, env.Value == "2100")
+
+		case "KIE_SERVER_KAFKA_EXT_CLIENT_ID":
+			assert.True(t, env.Value == "C1234567")
+
+		case "KIE_SERVER_KAFKA_EXT_SERVERS":
+			assert.True(t, env.Value == "localhost:9092")
+
+		case "KIE_SERVER_KAFKA_EXT_TOPICS":
+			assert.True(t, env.Value == "events=my-topics,errors=my-errs")
+		}
+	}
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_ENABLED"))
+	assert.Equal(t, "my-kafka-group", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_GROUP_ID"))
+	assert.Equal(t, "2", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_ACKS"))
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_AUTOCREATE_TOPICS"))
+	assert.Equal(t, "2100", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_MAX_BLOCK_MS"))
+	assert.Equal(t, "C1234567", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_CLIENT_ID"))
+	assert.Equal(t, "localhost:9092", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_SERVERS"))
+	assert.Equal(t, "events=my-topics,errors=my-errs", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_KAFKA_EXT_TOPICS"))
+}
+
+func createKafkaExtObject() *api.KafkaExtObject {
+	kafkaExtObject := api.KafkaExtObject{
+		MaxBlockMs:       Pint32(2100),
+		AutocreateTopics: Pbool(true),
+		BootstrapServers: "localhost:9092",
+		GroupID:          "my-kafka-group",
+		Acks:             Pint(2),
+		Topics:           []string{"events=my-topics", "errors=my-errs"},
+		ExtEnabled:       true,
+		ClientID:         "C1234567",
+	}
+	return &kafkaExtObject
+}
+
+func TestRhdmEnvironmentWithKafkaExtDefault(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
```

Comment:
we need to verify what happens w/ multiple servers configured... maybe set one w/ servers[0].name so you can track the configs better in status. the same kafka configs should stay put in status.

Reply:
Three servers, one with kafka default, one without kafka and one with kafka configured

Reply:
Three server, one with default, one without kafka and one configured, could be ok ?